File: tests/go/tests-go-testing-allure3-net_http/meta.go
package tests

import (
	"testing"

	allure "github.com/allure-framework/allure-go/commons/gotest"
)

func Wrap(t *testing.T, displayName string, body func(*allure.Context), opts ...allure.Option) {
	t.Helper()
	allure.Wrap(t, body, append([]allure.Option{allure.WithDisplayName(displayName)}, opts...)...)
}

func commonMeta() []allure.Option {
	return []allure.Option{
		allure.WithOwner("stanislav"),
		allure.WithLabel("module", "tests-go-testing-allure3-net_http"),
		allure.WithLabel("language", "go"),
	}
}

func LayerAPI(suite, epic, feature, severity string) []allure.Option {
	return append(commonMeta(),
		allure.WithSuite(suite),
		allure.WithEpic(epic),
		allure.WithFeature(feature),
		allure.WithSeverity(severity),
		allure.WithLabel("layer", "api"),
		allure.WithTag("api"),
		allure.WithLabel("framework", "net/http"),
	)
}

func LayerInfra(suite, epic, feature, severity string) []allure.Option {
	return append(commonMeta(),
		allure.WithSuite(suite),
		allure.WithEpic(epic),
		allure.WithFeature(feature),
		allure.WithSeverity(severity),
		allure.WithLabel("layer", "infra"),
		allure.WithTag("infra"),
		allure.WithTag("infra-backend"),
	)
}

func LayerManual(suite, epic, feature, severity string) []allure.Option {
	return append(commonMeta(),
		allure.WithSuite(suite),
		allure.WithEpic(epic),
		allure.WithFeature(feature),
		allure.WithSeverity(severity),
		allure.WithLabel("layer", "manual"),
		allure.WithTag("manual"),
		allure.WithLabel("ALLURE_MANUAL", "true"),
	)
}

File: tests/rust/tests-rust-testing-reqwest/src/meta.rs
use allure_rust_commons::{epic, feature, label, owner, severity, suite, tag};

use crate::config::ensure_allure_results_dir;

const MODULE: &str = "tests-rust-testing-reqwest";

fn common_meta() {
    ensure_allure_results_dir();
    owner("stanislav");
    label("module", MODULE);
    label("language", "rust");
}

pub fn layer_api(suite_name: &str, epic_name: &str, feature_name: &str, severity_name: &str) {
    common_meta();
    suite(suite_name);
    epic(epic_name);
    feature(feature_name);
    severity(severity_name);
    label("layer", "api");
    tag("api");
    label("framework", "reqwest");
}

pub fn layer_infra(suite_name: &str, epic_name: &str, feature_name: &str, severity_name: &str) {
    common_meta();
    suite(suite_name);
    epic(epic_name);
    feature(feature_name);
    severity(severity_name);
    label("layer", "infra");
    tag("infra");
    tag("infra-backend");
}
